src/Adapter/PlatformServices.Universal/Services/UniversalFileOperations.cs

```diff
@@ -74,8 +74,21 @@ public bool DoesFileExist(string assemblyFileName)
         /// </returns>
         public object CreateNavigationSession(string source)
         {
-            // Navigation session for UWP is handled by Desktop Adapter
-            return null;
+            var messageFormatOnException =
+                string.Join("MSTestDiscoverer:DiaSession: Could not create diaSession for source:", source, ". Reason:{0}");
+
+            try
+            {
+                Assembly objectModelAssembly = Assembly.Load(new AssemblyName("Microsoft.VisualStudio.TestPlatform.ObjectModel"));
+                Type diaType = objectModelAssembly.GetType("Microsoft.VisualStudio.TestPlatform.ObjectModel.DiaSession");
```

Comment:
Not needed, since this should be invoked only once when someone is trying to create DiaSesion, & then hold the session object.
Further the session is dependent based on for what exe/dll it is opened for, so caching would be wrong.

Reply:
It gets called for each assembly. I meant the GetType reflection call. You can still go ahead and create a new object each time.

Reply:
yes the type object I'll cache